examples/DataApi-Voting-poll/template/modules/shared_data/src/main/scala/com/my/currency/shared_data/LifecycleSharedFunctions.scala

```diff
@@ -0,0 +1,82 @@
+package com.my.currency.shared_data
+
+import cats.Monad
+import cats.conversions.all.autoWidenFunctor
+import cats.data.NonEmptyList
+import cats.effect.Async
+import cats.implicits.{catsSyntaxApplicativeId, catsSyntaxEitherId, catsSyntaxValidatedIdBinCompat0, toFlatMapOps, toFunctorOps}
+import com.my.currency.shared_data.Utils.getLastMetagraphIncrementalSnapshotInfo
+import org.tessellation.currency.dataApplication.dataApplication.DataApplicationValidationErrorOr
+import org.tessellation.currency.dataApplication.{DataApplicationValidationError, DataState, L0NodeContext, L1NodeContext}
+import org.tessellation.security.signature.Signed
+import com.my.currency.shared_data.combiners.Combiners.{combineCreatePoll, combineVoteInPoll}
+import com.my.currency.shared_data.errors.Errors.CouldNotGetLatestCurrencySnapshot
+import com.my.currency.shared_data.validations.Validations.{createPollValidations, createPollValidationsWithSignature, voteInPollValidations, voteInPollValidationsWithSignature}
+import com.my.currency.shared_data.types.Types.{CreatePoll, PollUpdate, VoteCalculatedState, VoteInPoll, VoteStateOnChain}
+import org.slf4j.LoggerFactory
+import org.tessellation.security.SecurityProvider
+
+object LifecycleSharedFunctions {
+
+  private val logger = LoggerFactory.getLogger("Data")
+
+  def validateUpdate[F[_] : Monad](update: PollUpdate)(implicit context: L1NodeContext[F]): F[DataApplicationValidationErrorOr[Unit]] = {
+    val lastCurrencySnapshot = context.getLastCurrencySnapshot
+
+    lastCurrencySnapshot.map(_.get.ordinal).flatMap { lastSnapshotOrdinal =>
+      update match {
+        case poll: CreatePoll =>
+          createPollValidations(poll, None, Some(lastSnapshotOrdinal))
+        case voteInPoll: VoteInPoll =>
+          getLastMetagraphIncrementalSnapshotInfo(context.asRight[L0NodeContext[F]]).flatMap {
+            case Some(snapshotInfo) => voteInPollValidations(voteInPoll, None, Some(lastSnapshotOrdinal), snapshotInfo)
+            case None => CouldNotGetLatestCurrencySnapshot.asInstanceOf[DataApplicationValidationError].invalidNec[Unit].pure[F]
+          }
+      }
+    }
```

Comment:
```suggestion
    context
      .getLastCurrencySnapshot
      .map(_.get.ordinal)
      .flatMap { lastSnapshotOrdinal =>
        update match {
          case poll: CreatePoll => createPollValidations(poll, none, lastSnapshotOrdinal.some)

          case voteInPoll: VoteInPoll =>
            getLastMetagraphIncrementalSnapshotInfo(context.asRight[L0NodeContext[F]])
              .flatMap {
                case Some(snapshotInfo) => voteInPollValidations(voteInPoll, none, lastSnapshotOrdinal.some, snapshotInfo)
                case None => CouldNotGetLatestCurrencySnapshot.invalidNec[Unit].pure[F]
              }
        }
      }
```
You'll need `import cats.syntax.option._` for the `none` and `.some` -- it's a good habit to use these since in certain circumstances they help the compiler with type inference.

Reply:
Got it updated